### ISSUE TITLE
New auth error codes missing in error object

### ISSUE THREAD

**Issue by melchiorb** [supabase/supabase-js#1023]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

The new error code (which is correctly sent by the server) is not accessible in auth-js.

Example when using non existend email address on signin:

Server response:
`{"code":"otp_disabled","message":"Signups not allowed for otp"}`

The error object returned by "supabase.auth.signInWithOtp":
`error.code = undefined`
`error.message = "Signups not allowed for otp"`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Trigger auth error
2. Inspect error object

## Expected behavior

`error.code = "otp_disabled"`

## System information

- Version of supabase-js: 2.42.5
- Version of auth-js: 2.63.1

**Comment by RSginer** [comment]
any updates on this ?

**Comment by Ahmed-seyam** [comment]
same issue here

**Comment by tom-at-pixel** [comment]
Seems to be working fine now. Confirm and close?

**Comment by sidharthv96** [comment]
This was working previously, but is exhibiting the same behaviour with code being undefined on `v2.47.10`, can confirm it's not working since `v2.46.2`.

**Comment by tom-at-pixel** [comment]
@sidharthv96 Can you try again and let us know if it's working now?

**Comment by michaelO93** [comment]
Still not working. 
Documentation says:

> If the user hasn't signed up yet, they are automatically signed up by default. To prevent this, set the `shouldCreateUser` option to false.
https://supabase.com/docs/guides/auth/auth-email-passwordless?queryGroups=language&language=js

But this is not the case. Always return "Signups not allowed for top" for non-registered emails.

**Comment by ghost** [comment]
I am having similar problems. I’m stuck on enabling magic-link sign-in for an existing user on my production project.
Endpoint: POST /auth/v1/otp?redirect\to=*****
Response: 422 Unprocessable Content with header x-sb-error-code: otp\disabled
{ "code": "otp_disabled", "message": "Signups not allowed for otp" }
What I’ve confirmed so far:
Auth → Settings → General
“Allow new users to sign up” = ON
Auth → Providers → Email
“Enable email provider” = ON
“Allow users to sign up with email” = ON
SMTP (SendGrid) test email succeeds
/auth/v1/settings shows
disable\signup: false
external.email: true
User I’m testing does exist and is confirmed in auth.users.
Front-end call (Nuxt 3):
await supabase.auth.signInWithOtp({ email, options: { shouldCreateUser: false, emailRedirectTo: '*****' }});\
I redeployed and waited several minutes after toggling settings, but still get otp\disabled every time.
Any idea what else could cause the OTP endpoint to return otp\disabled?
Happy to share more logs if helpful—thanks!

**Comment by onurusluca** [comment]
This is still happening. Any updates?

**Comment by giangchau92** [comment]
Same issue here? Any update?

**Comment by onurusluca** [comment]
I don't know how but after restarting the server and adding Google Auth, OTP started working. I wasn't even trying to do anything.

But the problem is now that if someone tries to login with an email that doesn't exist in the database, I get otp is diabled error. Things are weird, I don't know.

**Comment by giangchau92** [comment]
Yes, some said send OTP (email or sms) are not allow if user not existed. So, I need to signUp first then send OTP start working

**Comment by onurusluca** [comment]
@giangchau92 The thing is, it should return something like email doesn't exist or similar. This way, there is no knowing and we can't tell the user what is happening. I am thinking Supabase OTP is not a good idea to use at this stage. And it looks like it won't be fixed because this problem is more than 2 years old.
